### DISCUSSION TITLE
Is there an option to automatically integrate the metadata.timestamp?

### DISCUSSION THREAD

**Discussion by seahr** [CycloneDX/cyclonedx-php-composer#327] [Q&A] [answered]
Part of the [The Minimum Elements For a Software Bill of Materials (SBOM)](https://www.ntia.doc.gov/files/ntia/publications/sbom_minimum_elements_report.pdf) is the Timestamp (Record of the date and time of the SBOM data assembly.; cf. page 9 of PDF), which makes a lot of sense: 

> Timestamp records when the data is assembled -- the point of the SBOM creation. These further support the origin of the data, and help identify updated versions of the SBOM. These data fields provide context to the SBOM data source, and can potentially be used to make trust determinations.

It is specified in the CycloneDX standard ([XSD](https://github.com/CycloneDX/specification/blob/1.3/schema/bom-1.3.xsd), cf. line 40 ff.) as:

`<xs:element name="timestamp" type="xs:dateTime" minOccurs="0"><xs:annotation><xs:documentation>The date and time (timestamp) when the document was created.</xs:documentation></xs:annotation><xs:element>`

Is there an option to automatically insert the timestamp as part of the metadata? Should we enable this as a feature?

If not: How do you handle timestamps while automatically generating sboms?

**Comment by jkowalleck** [comment, marked as answer]
it is already part of the [next major version V4](https://github.com/CycloneDX/cyclonedx-php-composer/milestone/5), which is in the pipeline.

see #112

missing the feature badly? feel free to implement it into the current version and pull request it.

**Reply by jkowalleck** [reply]
v4 was just released: https://github.com/CycloneDX/cyclonedx-php-composer/releases/tag/v4.0.0
